torch/csrc/jit/ir.cpp

```diff
@@ -1258,7 +1258,14 @@ Node* Graph::createTupleSlice(Value* tup, int64_t beg, int64_t end) {
 Node* Graph::createList(const TypePtr& elem_type, at::ArrayRef<Value*> values) {
   auto n = create(prim::ListConstruct, values);
   for (const auto& v : values) {
-    AT_ASSERT(v->type()->isSubtypeOf(elem_type));
+    AT_CHECK(
```

Comment:
While this could make some error messages better, this is making the error appear to be a User error when it in fact is a developer error. The user-facing error is already supposed to be checked here [here](https://github.com/pytorch/pytorch/blob/4d676d53a67e610689afbb34deabcac6ba8b8c50/torch/csrc/jit/script/compiler.cpp#L2515). I don't think we should have two user-facing errors for the same concept.

The only other time i've seen this tripped up is because of an error when trying to match a VarType schema - which is a Developer error.